src/material/datepicker/date-selection-model.ts

```diff
@@ -69,8 +72,9 @@ export abstract class MatDateSelectionModel<S, D = ExtractDateTypeFromSelection<
    * @param source Object that triggered the selection change.
    */
   updateSelection(value: S, source: unknown) {
+    const oldValue = (this as {selection: S}).selection;
     (this as {selection: S}).selection = value;
-    this._selectionChanged.next({selection: value, source});
+    this._selectionChanged.next({selection: value, source, oldValue});
```

Comment:
Would it make sense to check whether the selection has changed here before emitting the event? That way we can avoid having to attach the `oldValue`.

Reply:
I don't think that would work since the same selection change is fired when either the start date or the end date is changed.

Reply:
What I meant is that it could look at the old start and end dates and only emit if either one has changed.

Reply:
It's already doing that. The additional issue I'm trying to fix is to not have the start date fire a change when the end date has changed and vice versa.